internal/provider/backups_data_source_test.go

```diff
@@ -0,0 +1,202 @@
+		RetentionDays:    30,
+	}
+
+	now := time.Now()
+	// Set endTime and startTime so that they have 0 for nanoseconds.
+	// This will ensure that tests pass when checking against the time.RFC3339 format.
+	endTime := time.Date(now.Year(), now.Month(), now.Day(), now.Hour(), now.Minute(), now.Second(), 0, time.UTC).Add(24 * time.Hour)
+	startTime := endTime.Add(-3 * 24 * time.Hour)
+	// Set endDate and startDate so that they only have the year, month, and day populated.
+	// This will ensure that tests pass when checking against the YYYY-MM-DD format.
+	endDate := time.Date(endTime.Year(), endTime.Month(), endTime.Day(), 0, 0, 0, 0, time.UTC)
+	startDate := time.Date(startTime.Year(), startTime.Month(), startTime.Day(), 0, 0, 0, 0, time.UTC)
+	limit := int32(2)
+	order := "DESC"
+
+	listBackupsOptionsWithDate := &client.ListBackupsOptions{
+		StartTime:           &startDate,
+		EndTime:             &endDate,
+		PaginationLimit:     &limit,
+		PaginationSortOrder: &order,
+	}
+	listBackupsOptionsWithTimestamp := &client.ListBackupsOptions{
+		StartTime:           &startTime,
+		EndTime:             &endTime,
+		PaginationLimit:     &limit,
+		PaginationSortOrder: &order,
+	}
+
+	// Mock backups response.
+	backups := &client.ListBackupsResponse{
+		Backups: []client.BackupSummary{
+			{
+				Id:       "00000000-0000-0000-0000-000000000001",
+				AsOfTime: time.Date(2023, 1, 1, 0, 0, 0, 0, time.UTC),
+			},
+		},
+	}
+
+	httpOkResponse := &http.Response{Status: http.StatusText(http.StatusOK)}
+
+	s.EXPECT().CreateCluster(gomock.Any(), gomock.Any()).Return(cluster, nil, nil)
+	s.EXPECT().GetCluster(gomock.Any(), clusterID).Return(cluster, httpOkResponse, nil).AnyTimes()
+	s.EXPECT().UpdateBackupConfiguration(gomock.Any(), clusterID, gomock.Any()).Return(backupConfig, httpOkResponse, nil)
+	s.EXPECT().GetBackupConfiguration(gomock.Any(), clusterID).Return(backupConfig, httpOkResponse, nil).AnyTimes()
+	s.EXPECT().ListBackups(gomock.Any(), clusterID, listBackupsOptionsWithDate).Return(backups, httpOkResponse, nil).Times(6)
+	s.EXPECT().ListBackups(gomock.Any(), clusterID, listBackupsOptionsWithTimestamp).Return(backups, httpOkResponse, nil).AnyTimes()
+	s.EXPECT().DeleteCluster(gomock.Any(), clusterID)
+
+	ctx := context.Background()
+	testBackupsDataSource(
+		t,
+		clusterName,
+		startTime.Format(time.RFC3339),
+		endTime.Format(time.RFC3339),
+		true,
+		s,
+		ctx,
+	)
+}
+
+func testBackupsDataSource(
+	t *testing.T,
+	clusterName string,
+	startTime string,
+	endTime string,
+	useMock bool,
+	service client.Service,
+	ctx context.Context,
+) {
+	backupsDataSourceName := "data.cockroach_backups.test"
+
+	resource.Test(t, resource.TestCase{
+		IsUnitTest:               useMock,
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: getTestBackupsDataSourceConfig(clusterName, startTime[:10], endTime[:10]),
```

Comment:
Why do you need these string modifications?  `:10`.  Can you make this a bit clearer?